packages/odata-service-inquirer/src/utils/index.ts

```diff
@@ -70,6 +83,24 @@ function createTelemetryEvent(eventName: string, telemetryData: TelemetryPropert
     };
 }
 
+/**
+ * Used only to generate telemetry events in the case of destination errors.
+ *
+ * @param destination
+ * @returns the telemetry property destination type
+ */
+export function getTelemPropertyDestinationType(destination: Destination): TelemPropertyDestinationType {
+    if (isAbapODataDestination(destination)) {
+        return 'AbapODataCatalogDest';
+    } else if (isFullUrlDestination(destination)) {
+        return 'GenericODataFullUrlDest';
+    } else if (isPartialUrlDestination(destination)) {
+        return 'GenericODataPartialUrlDest';
+    } else {
+        return 'Unknown';
```

Comment:
Obviously there can be other other types i.e. `abap_cloud` but do we care for this flow?